### ISSUE TITLE
hubot unresponsive in channel after upgrade

### ISSUE THREAD

**Issue by linxcat** [loafoe/hubot-matteruser#53]
After upgrading hubot to a more recent matteruser driver, the bot stopped replying to channel mentions, but still responds when sent a DM.

"hubot-matteruser": "4.2.0",
"mattermost-client": "4.2.0",

using 3.9.1 and 3.9.0 the bot replies as expected.

This is all taking place on mattermost 4.1.0

Using debugging I can see that Hubot receives my message and schedules a reply, but the reply never comes to the channel

**Comment by linxcat** [comment]
we believed it was an api v4 problem, but upgrading to v 3.10.0 does not reintroduce the problem, I am checking which patch will cause the problem.

**Comment by linxcat** [comment]
confirmed first 4.0 release to cause the problem

**Comment by loafoe** [comment]
@linxcat I will investigate. Have been developing against 4.1.x and 4.2.x servers for the last days. Can you send me a list of all `MATTERMOST_` config variables that are active? That will help reproduce the conditions. thx!

**Comment by kuuji** [comment]
Hey @loafoe we actually discovered what it was yesterday night. It turns out that you stopped overriding hubot name by the mattermost username in this commit => https://github.com/loafoe/hubot-matteruser/commit/7964bc61d160db2f0c953f81cefe5b9340e0f37b.

And of course, we were using a different name than the mattermost username when we started hubot. It's all good now and we can close this.

Oh and btw, great work on this adapter, we've been using it for quite some time now and it works like a charm.
